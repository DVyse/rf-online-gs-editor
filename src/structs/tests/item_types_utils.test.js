import {
  PREFIXES,
  FINITES,
  CONV_DECIMALS,
  getFiniteByTypeName,
  getTypeNameByFinite,
  getPrefixByTypeName,
  getTypeNamesByPrefix,
  getConvDecimalByTypeName,
} from '../item_types_utils';

describe('Structs', () => {
  describe('ItemTypesUtils', () => {
    it('should have prefixes', () => {
      expect(PREFIXES).toMatchSnapshot();
    });
    it('should have finites', () => {
      expect(FINITES).toMatchSnapshot();
    });
    it('should have convDecimals', () => {
      expect(CONV_DECIMALS).toMatchSnapshot();
    });
    it('should return finite by typeName', () => {
      expect(getFiniteByTypeName('upper')).toEqual(1);
      expect(getFiniteByTypeName('cloak')).toEqual(8);
      expect(getFiniteByTypeName('___undefined___')).toEqual(undefined);
    });
    it('should return typeName by finite', () => {
      expect(getTypeNameByFinite(1)).toEqual('upper');
      expect(getTypeNameByFinite(8)).toEqual('cloak');
      expect(getTypeNameByFinite('___undefined___')).toEqual(undefined);
    });
    it('should return prefix by typeName', () => {
      expect(getPrefixByTypeName('upper')).toEqual('iu');
      expect(getPrefixByTypeName('cloak')).toEqual('ik');
      expect(getPrefixByTypeName('___undefined___')).toEqual(undefined);
    });
    it('should return typeNames by prefix', () => {
      expect(getTypeNamesByPrefix('iu')).toMatchSnapshot();
      expect(getTypeNamesByPrefix('ik')).toMatchSnapshot();
      expect(getTypeNamesByPrefix('un')).toMatchSnapshot();
      expect(getTypeNamesByPrefix('___undefined___')).toEqual([]);
    });
    it('should return typeNames by prefix', () => {
      expect(getConvDecimalByTypeName('upper')).toEqual(192);
      expect(getConvDecimalByTypeName('cloak')).toEqual(192);
      expect(getConvDecimalByTypeName('coupon')).toEqual(96);
      expect(getConvDecimalByTypeName('___undefined___')).toEqual(undefined);
    });
  });
});
