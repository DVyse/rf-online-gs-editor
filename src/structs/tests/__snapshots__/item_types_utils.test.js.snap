// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Structs ItemTypesUtils should have convDecimals 1`] = `
Object {
  "amulet": 192,
  "animus": 192,
  "bag": 192,
  "battery": 192,
  "booty": 192,
  "box": 80,
  "bullet": 192,
  "cashmining": 128,
  "cloak": 192,
  "coupon": 96,
  "dungeonkey": 192,
  "face": 192,
  "firecracker": 144,
  "force": 192,
  "gauntlet": 192,
  "helmet": 192,
  "lower": 192,
  "maketool": 192,
  "map": 192,
  "mauarm": 128,
  "mauback": 128,
  "maubullet": 128,
  "mauhead": 128,
  "maulower": 128,
  "maushoulder": 128,
  "mauupper": 128,
  "ore": 192,
  "pager": 240,
  "potion": 192,
  "quest": 128,
  "radar": 80,
  "recovery": 80,
  "resource": 192,
  "ring": 192,
  "shield": 192,
  "shoe": 192,
  "siegekit": 96,
  "ticket": 112,
  "tower": 160,
  "town": 192,
  "trap": 112,
  "unitkey": 192,
  "upper": 192,
  "weapon": 192,
}
`;

exports[`Structs ItemTypesUtils should have finites 1`] = `
Object {
  "amulet": 10,
  "animus": 24,
  "bag": 14,
  "battery": 15,
  "booty": 20,
  "box": 31,
  "bullet": 11,
  "cashmining": 33,
  "cloak": 8,
  "coupon": 36,
  "dungeonkey": 23,
  "face": 0,
  "firecracker": 32,
  "force": 18,
  "gauntlet": 3,
  "helmet": 5,
  "lower": 2,
  "maketool": 12,
  "map": 21,
  "mauarm": 40,
  "mauback": 42,
  "maubullet": 43,
  "mauhead": 37,
  "maulower": 39,
  "maushoulder": 41,
  "mauupper": 38,
  "ore": 16,
  "pager": 35,
  "potion": 13,
  "quest": 29,
  "radar": 34,
  "recovery": 30,
  "resource": 17,
  "ring": 9,
  "shield": 7,
  "shoe": 4,
  "siegekit": 27,
  "ticket": 28,
  "tower": 25,
  "town": 22,
  "trap": 26,
  "unitkey": 19,
  "upper": 1,
  "weapon": 6,
}
`;

exports[`Structs ItemTypesUtils should have prefixes 1`] = `
Object {
  "amulet": "ia",
  "animus": "ij",
  "bag": "ie",
  "battery": "it",
  "booty": "iy",
  "box": "bx",
  "bullet": "ib",
  "cashmining": "un",
  "cloak": "ik",
  "coupon": "cu",
  "dungeonkey": "ix",
  "face": "if",
  "firecracker": "fi",
  "force": "ic",
  "gauntlet": "ig",
  "helmet": "ih",
  "lower": "il",
  "maketool": "im",
  "map": "iz",
  "mauarm": "un",
  "mauback": "un",
  "maubullet": "un",
  "mauhead": "un",
  "maulower": "un",
  "maushoulder": "un",
  "mauupper": "un",
  "ore": "io",
  "pager": "lk",
  "potion": "ip",
  "quest": "ev",
  "radar": "rd",
  "recovery": "re",
  "resource": "ir",
  "ring": "ii",
  "shield": "id",
  "shoe": "is",
  "siegekit": "sk",
  "ticket": "ti",
  "tower": "gt",
  "town": "iq",
  "trap": "tr",
  "unitkey": "in",
  "upper": "iu",
  "weapon": "iw",
}
`;

exports[`Structs ItemTypesUtils should return typeNames by prefix 1`] = `
Array [
  "upper",
]
`;

exports[`Structs ItemTypesUtils should return typeNames by prefix 2`] = `
Array [
  "cloak",
]
`;

exports[`Structs ItemTypesUtils should return typeNames by prefix 3`] = `
Array [
  "cashmining",
  "mauhead",
  "mauupper",
  "maulower",
  "mauarm",
  "maushoulder",
  "mauback",
  "maubullet",
]
`;
